app/src/androidTest/java/au/com/shiftyjelly/pocketcasts/repositories/sync/PodcastSyncProcessTest.kt

```diff
@@ -0,0 +1,234 @@
+package au.com.shiftyjelly.pocketcasts.repositories.sync
+
+import android.content.Context
+import androidx.room.Room
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import au.com.shiftyjelly.pocketcasts.models.db.AppDatabase
+import au.com.shiftyjelly.pocketcasts.models.entity.PodcastEpisode
+import au.com.shiftyjelly.pocketcasts.preferences.AccessToken
+import au.com.shiftyjelly.pocketcasts.preferences.Settings
+import au.com.shiftyjelly.pocketcasts.repositories.bookmark.BookmarkManagerImpl
+import au.com.shiftyjelly.pocketcasts.repositories.podcast.EpisodeManager
+import au.com.shiftyjelly.pocketcasts.repositories.podcast.FolderManager
+import au.com.shiftyjelly.pocketcasts.repositories.podcast.PlaylistManager
+import au.com.shiftyjelly.pocketcasts.repositories.podcast.PodcastManager
+import au.com.shiftyjelly.pocketcasts.repositories.user.StatsManager
+import au.com.shiftyjelly.pocketcasts.servers.di.ServersModule
+import au.com.shiftyjelly.pocketcasts.servers.sync.SyncServerManager
+import au.com.shiftyjelly.pocketcasts.utils.extensions.toIsoString
+import junit.framework.TestCase.assertEquals
+import junit.framework.TestCase.assertNull
+import kotlinx.coroutines.test.runTest
+import okhttp3.Cache
+import okhttp3.OkHttpClient
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.Assert.assertNotNull
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.whenever
+import retrofit2.Retrofit
+import timber.log.Timber
+import java.net.HttpURLConnection
+import java.util.Date
+import java.util.concurrent.TimeUnit
+
+@RunWith(AndroidJUnit4::class)
+class PodcastSyncProcessTest {
+
+    private lateinit var context: Context
+    private lateinit var mockWebServer: MockWebServer
+    private lateinit var retrofit: Retrofit
+    private lateinit var okhttpCache: Cache
+    private lateinit var appDatabase: AppDatabase
+
+    @Before
+    fun setUp() {
+        context = InstrumentationRegistry.getInstrumentation().targetContext
+
+        mockWebServer = MockWebServer()
+        mockWebServer.start()
+
+        appDatabase = Room.inMemoryDatabaseBuilder(context, AppDatabase::class.java).build()
+
+        val moshi = ServersModule.provideMoshiBuilder().build()
+        val okHttpClient = OkHttpClient.Builder().build()
+        retrofit = ServersModule.provideRetrofit(baseUrl = mockWebServer.url("/").toString(), okHttpClient = okHttpClient, moshi = moshi)
+        okhttpCache = ServersModule.provideCache(folder = "TestCache", context = context)
+    }
+
+    /**
+     * Test the sync process handles bookmarks.
+     */
+    @Test
+    fun bookmarkSync() {
```

Comment:
Great job setting it up! 👏 

Can we split it into one each for updating, inserting, and deleting a bookmark so that it is more readable?